core/src/main/java/org/jruby/RubyClass.java

```diff
@@ -484,7 +485,11 @@ static boolean notVisibleAndNotMethodMissing(DynamicMethod method, String name,
     }
 
     public IRubyObject finvoke(ThreadContext context, IRubyObject self, String name, Block block) {
-        CacheEntry entry = searchWithCache(name);
+        return finvokeWithRefinements(context, self, null, name, block);
+    }
+
+    public IRubyObject finvokeWithRefinements(ThreadContext context, IRubyObject self, StaticScope staticScope, String name, Block block) {
```

Comment:
If all finvoke calls *withRefinements should we just put the body of *withRefinements into finvoke and not have these extra signatures?

Reply:
It's a signature issue; a new signature is needed to pass in the scope to use for refined lookup.